pkg/providers/imagefamily/azlinux.go

```diff
@@ -97,7 +97,6 @@ func (u AzureLinux) UserData(kubeletConfig *corev1beta1.KubeletConfiguration, ta
 		ClusterID:                      u.Options.ClusterID,
 		APIServerName:                  u.Options.APIServerName,
 		KubeletClientTLSBootstrapToken: u.Options.KubeletClientTLSBootstrapToken,
-		NetworkPlugin:                  u.Options.NetworkPlugin,
```

Comment:
Hello, after reading your PR, I think this is a great idea. However, I have a question as to why this line needs to be removed. I went to look at the provider.state.sh  of Azure VM, which determines whether to install Azure CNI based on the ` NETWORK-POLICY `. If this line is removed, will it cause Azure CNI installation to fail?

Reply:
installNetworkPlugin() {
    if [[ "${NETWORK_PLUGIN}" = "azure" ]]; then
        installAzureCNI
    fi
    installCNI #reference plugins. Mostly for kubenet but loopback plugin is used by containerd until containerd 2
    rm -rf $CNI_DOWNLOADS_DIR & 
}

Reply:
Good catch you are correct, I discovered similar problems when testing it this morning, seems you are pointing to the exact sad part in agentbaker.

Reply:
:tada: aks-general-purpose-p7wg4           Ready    <none>   31s   v1.29.7

Nodes come up as ready after commit `5d88685` with fully initialized cni.